src/WebJobs.Script/Diagnostics/Extensions/ScriptHostLoggerExtension.cs

```diff
@@ -0,0 +1,174 @@
+            LoggerMessage.Define<string>(
+            LogLevel.Error,
+            new EventId(0, nameof(FunctionsErrors)),
+            "{message}");
+
+        private static readonly Action<ILogger, Exception> _addingDescriptorProvidersForAllLanguages =
+            LoggerMessage.Define(
+            LogLevel.Debug,
+            new EventId(0, nameof(AddingDescriptorProvidersForAllLanguages)),
+            "Adding Function descriptor providers for all languages.");
+
+        private static readonly Action<ILogger, string, Exception> _addingDescriptorProviderForLanguage =
+            LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(0, nameof(AddingDescriptorProviderForLanguage)),
+            "Adding Function descriptor provider for language {workerRuntime}.");
+
+        private static readonly Action<ILogger, Exception> _creatingDescriptors =
+            LoggerMessage.Define(
+            LogLevel.Debug,
+            new EventId(0, nameof(CreatingDescriptors)),
+            "Creating function descriptors.");
+
+        private static readonly Action<ILogger, Exception> _descriptorsCreated =
+            LoggerMessage.Define(
+            LogLevel.Debug,
+            new EventId(0, nameof(DescriptorsCreated)),
+            "Function descriptors created.");
+
+        private static readonly Action<ILogger, Exception> _errorPurgingLogFiles =
+            LoggerMessage.Define(
+            LogLevel.Error,
+            new EventId(0, nameof(ErrorPurgingLogFiles)),
+            "An error occurred while purging log files");
+
+        private static readonly Action<ILogger, string, Exception> _deletingLogDirectory =
+            LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(0, nameof(DeletingLogDirectory)),
+            "Deleting log directory '{logDir}'");
+
+        private static readonly Action<ILogger, string, string, Exception> _failedToLoadType =
+            LoggerMessage.Define<string, string>(
+            LogLevel.Warning,
+            new EventId(0, nameof(FailedToLoadType)),
+            "Failed to load type '{typeName}' from '{path}'");
+
+        private static readonly Action<ILogger, string, Exception> _configurationError =
+            LoggerMessage.Define<string>(
+            LogLevel.Information,
+            new EventId(0, nameof(ConfigurationError)),
+            "{message}");
+
+        private static readonly Action<ILogger, string, Exception> _versionRecommendation =
+            LoggerMessage.Define<string>(
+            LogLevel.Warning,
+            new EventId(0, nameof(VersionRecommendation)),
+            "{message}");
+
+        private static readonly Action<ILogger, string, Exception> _scriptHostInitialized =
+            LoggerMessage.Define<string>(
+            LogLevel.Information,
+            new EventId(0, nameof(ScriptHostInitialized)),
+            "Host initialized ({msString}ms)");
```

Comment:
Can you make this `LoggerMessage.Define<long>` so we don't have to call `ToString()` on it before passing it in? I see that in a couple places.